recipe/install_nvcc.sh

```diff
@@ -2,61 +2,66 @@
 
 set -xeuo pipefail
 
-# Default to using `nvcc` to specify `CUDA_HOME`.
-if [ -z ${CUDA_HOME+x} ]
-then
-    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
-fi
-
 # Set `CUDA_HOME` in an activation script.
 mkdir -p "${PREFIX}/etc/conda/activate.d"
-cat > "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
+cat >"${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
 #!/bin/bash
 
-if [[ ! -d "${CUDA_HOME}" ]]
+# Default to using \$(cuda-gdb) to specify \$(CUDA_HOME).
+if [ -z \${CUDA_HOME+x} ]
+then
+    CUDA_HOME="\$(dirname \$(dirname \$(which cuda-gdb)))"
+fi
+
+if [[ ! -d "\${CUDA_HOME}" ]]
 then
-    echo "Cannot find: ${CUDA_HOME}"
+    echo "Cannot find: \${CUDA_HOME}"
     exit 1
 fi
-if [[ ! -f "${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
+if [[ ! -f "\${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
 then
-    echo "Cannot find: ${CUDA_HOME}/lib64/stubs/libcuda.so"
+    echo "Cannot find: \${CUDA_HOME}/lib64/stubs/libcuda.so"
     exit 1
 fi
-grep "CUDA Version ${PKG_VERSION}" ${CUDA_HOME}/version.txt &>/dev/null
-if [[ $? -ne 0 ]]
+grep "CUDA Version ${PKG_VERSION}" \${CUDA_HOME}/version.txt &>/dev/null
+if [[ \$? -ne 0 ]]
 then
     echo "Version of installed CUDA didn't match package"
     exit 1
 fi
 
-export CUDA_HOME="${CUDA_HOME}"
+export CUDA_HOME="\${CUDA_HOME}"
 export CFLAGS="\${CFLAGS} -I\${CUDA_HOME}/include"
 export CPPFLAGS="\${CPPFLAGS} -I\${CUDA_HOME}/include"
 export CXXFLAGS="\${CXXFLAGS} -I\${CUDA_HOME}/include"
+
+# Add \$(libcuda.so) shared object stub to the compiler sysroot.
+# Needed for things that want to link to $(libcuda.so).
+# Stub is used to avoid getting driver code linked into binaries.
+
+CONDA_ENV_SYSROOT="\$(\${CC} --print-sysroot)"
+mkdir -p "\${CONDA_ENV_SYSROOT}/lib"
+ln -s "\${CUDA_HOME}/lib64/stubs/libcuda.so" "\${CONDA_ENV_SYSROOT}/lib/libcuda.so"
 EOF
 
 # Unset `CUDA_HOME` in a deactivation script.
 mkdir -p "${PREFIX}/etc/conda/deactivate.d"
-cat > "${PREFIX}/etc/conda/deactivate.d/${PKG_NAME}_deactivate.sh" <<EOF
+cat >"${PREFIX}/etc/conda/deactivate.d/${PKG_NAME}_deactivate.sh" <<EOF
 #!/bin/bash
 unset CUDA_HOME
```

Comment:
@kkraus14, this needs to be removed.

Reply:
If a user didn't have `CUDA_HOME` set before the activate script than this should be unset. I think we just need some additional logic to track if CUDA_HOME was set before the activate script to know whether we should unset it.